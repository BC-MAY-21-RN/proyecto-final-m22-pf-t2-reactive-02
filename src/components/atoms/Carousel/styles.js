import {StyleSheet, Dimensions} from 'react-native';
import colors from '../../../const/colors';

const styles = (i, p) =>
  StyleSheet.create({
    container: {
      marginLeft: -16,
      height: i,
      width: Dimensions.get('screen').width - 8,
      backgroundColor: colors.black,
    },
    img: {
      height: 200,
      width: Dimensions.get('screen').width - 8,
      resizeMode: 'cover',
    },
    pager: {flex: 1},
    pagination: {
      flexDirection: 'row',
      position: 'absolute',
      bottom: 0,
      alignSelf: 'center',
    },
    point: {
      color: i === p ? 'white' : '#888',
      margin: 3,
    },
  });

export default styles;
