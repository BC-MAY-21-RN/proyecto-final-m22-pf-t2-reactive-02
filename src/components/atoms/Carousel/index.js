import React, {useState} from 'react';
import {View, Text, Image, TouchableOpacity} from 'react-native';
import PagerView from 'react-native-pager-view';
import styles from './styles';

const showImages = (index, changei, changev) => {
  changei(index);
  changev(true);
};

export default function Carousel({array, changev, changei}) {
  const [position, setPosition] = useState(0);
  const newPos = e => setPosition(e.nativeEvent.position);
  return (
    <View style={styles().container}>
      <PagerView onPageScroll={e => newPos(e)} style={styles().pager}>
        {array.map((url, index) => (
          <View key={index}>
            <TouchableOpacity
              onPress={() => showImages(position, changei, changev)}>
              <Image source={{uri: url}} style={styles().img} />
            </TouchableOpacity>
          </View>
        ))}
      </PagerView>
      <View style={styles().pagination}>
        {array.map((_, k) => (
          <Text key={k} style={styles(k, position).point}>
            â¬¤
          </Text>
        ))}
      </View>
    </View>
  );
}
