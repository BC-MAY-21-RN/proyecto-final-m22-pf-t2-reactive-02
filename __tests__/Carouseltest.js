import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Carousel from '../src/components/atoms/Carousel';

describe('Test from Carousel component', () => {
  it('Carousel snapshot test', () => {
    const renderCarousel = renderer
      .create(
        <Carousel
          array={[]}
          handleImgsVisible={() => {}}
          handleUrlImgs={() => {}}
        />,
      )
      .toJSON();
    expect(renderCarousel).toMatchSnapshot();
  });
});
