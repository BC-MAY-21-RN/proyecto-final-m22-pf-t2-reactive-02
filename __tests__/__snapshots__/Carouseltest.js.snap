// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from Carousel component Carousel snapshot test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "height": 0,
      "marginLeft": -16,
      "width": 742,
    }
  }
>
  <RNCViewPager
    layoutDirection="ltr"
    onMoveShouldSetResponderCapture={[Function]}
    onPageScroll={[Function]}
    onPageScrollStateChanged={[Function]}
    onPageSelected={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "bottom": 0,
        "flexDirection": "row",
        "position": "absolute",
      }
    }
  />
</View>
`;
